src/node/rpc/frontend.h

```diff
@@ -539,17 +570,31 @@ namespace ccf
                 ctx->set_response_headers(
                   GLOBAL_COMMIT, consensus->get_commit_seqno());
 
-                if (
-                  history && consensus->is_primary() &&
-                  (cv % sig_max_tx == sig_max_tx / 2))
                 {
-                  if (consensus->type() == ConsensusType::Raft)
+                  TryLock l(sign_lock);
```

Comment:
Do we need to lock or is it ok if we occasionally send out signatures more often?